### ISSUE TITLE
OptionalUnit for open methods

### ISSUE THREAD

**Issue by WildOrangutan** [detekt/detekt#4805]
## Expected Behavior
OptionalUnit rule to not detect optional `open` methods.

## Observed Behavior
OptionalUnit detected open method, that is used in abstract class.

In following example, this rule reports error for `onScream()`.

```
abstract class Monkey {
    
    /** Optional method, override if needed */
    open fun onScream() = Unit
    
    fun scream() {
        onScream()
    }
}
```
PS: Dumb example, but you get the point.

## Your Environment
* Detekt: 1.20.0

**Comment by BraisGabin** [comment]
I'm closing this because no answer was received to the Nico's proposal. Don't hestiate to reopen/comment if you think that we are missing something here.

**Comment by WildOrangutan** [comment]
Imho, I still consider it a bug. I think this rule shouldn't trigger for this case, since `Unit` is not unnecessary in this case.
